### PR TITLE
Add basic Emacs `.gitignore` entries

### PR DESCRIPTION
**Pull request by adamscott**
Adds these entries to the `.gitignore` file for the [Emacs editor](https://www.gnu.org/software/emacs/).
```
\#*#
.\#*
```
These are temporary files.

### CODE REVIEW

**Review by AThousandShips — APPROVED** [review]
Looked and confirmed these are in fact Emacs temporary files (I believe it also uses `~` variations which we already catch)

Showing off that Emacs nerd cred! /j (joke as in fake implying the existing attitudes of Emacs etc. and the editor wars, to clarify)

File: .gitignore [hunk 1]
```diff
@@ -132,6 +132,10 @@ cppcheck-cppcheck-build-dir/
 *.pydevproject
 *.launch
 
+# Emacs
+\#*#
```
Comment (aaronfranke) [inline .gitignore:136]: Doesn't this need to be `\#*\#` for the second `#` to be accepted? Otherwise that's a comment.

Comment (adamscott) [reply .gitignore:136]: Oh, right!

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.
Cherry-picked for 4.1.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.6.
Cherry-picked for 3.5.4.

**Pull request merged**